com.unity.perception/Documentation~/FAQ/FAQ.md

```diff
@@ -0,0 +1,584 @@
+    {
+        m_GameObjectOneWayCache.ResetAllObjects();
+    }
+
+    /// <summary>
+    /// Calculates the separation distance needed between placed objects to be sure that no two objects will overlap
+    /// </summary><br>
+    /// <returns>The max separation distance</returns>
+    float CalculateMaxSeparationDistance(ICollection<GameObject> categories)
+    {
+        var maxBoundsSize = m_GameObjectBoundsSizeCache.Where(item => categories.Contains(item.Key)).Max(pair => pair.Value); 
+        return maxBoundsSize * m_ScaleFactor * nonOverlapGuarantee;
+    }
+
+    float CalculateAverageBoundsSize()
+    {
+        foreach (var category in prefabs.categories)
+        {
+            var prefab = category.Item1;
+
+            prefab.transform.localScale = Vector3.one;
+            var renderers = prefab.GetComponentsInChildren<Renderer>();
+            var totalBounds = new Bounds();
+            foreach (var renderer in renderers)
+            {
+                totalBounds.Encapsulate(renderer.bounds);
+            }
+
+            var boundsSize = totalBounds.size.magnitude;
+            m_GameObjectBoundsSizeCache.Add(prefab, boundsSize);
+        }
+
+        return m_GameObjectBoundsSizeCache.Values.Average();
+    }
+}
+
+  ```
+---
+</details>
+
+<details>
+  <summary><strong>Q: What if I don't want randomized object placement? Can I move my objects in a non-randomized deterministic manner on each frame?</strong> </summary><br>
+
+Even though we call them Randomizers, you can use a Randomizer to perform any task through-out the execution lifecycle of your Scenario. The power of the Randomizers comes from the lifecycle hooks that they have into the Iteration and the Scenario, making it easy to know and guarantee when and in which order in the life of your simulation each piece of code runs. These functions include:
+* `OnEnable`
+* `OnAwake`
+* `OnUpdate`
+* `OnIterationStart`
+* `OnIterationEnd`
+* `OnScenarioStart`
+* `OnScenarioComplete`
+* `OnDisable`
+
+So, in order to have deliberate non-random object movement, you will just need to put your object movement code inside of one of the recurrent lifecycle functions. `OnUpdate()` runs on every frame of the simulation, and `OnIterationStart()` runs every Iteration (which can be the same as each frame if you have only 1 frame per Iteration of your Scenario). For example, the code below moves all objects tagged with the component `ForwardMoverTag` by along their forward axis by 1 unit, on every Iteration. 
+
+```C#
+protected override void OnIterationStart()
+{
+    var tags = tagManager.Query<ForwardMoverTag>();
+    foreach (var tag in tags)
+    {
+        tag.transform.Translate(Vector3.forward);
+    }
+}
+```
+---
+</details>
+
+<details>
+  <summary><strong>Q: The objects instantiated using the sample foreground placement Randomizer are floating in the air. How can I use this Randomizer to place objects on a horizontal surface instead?</strong> </summary><br>
```

Comment:
One more method of note for placing objects on a horizontal surface is to create a custom placement randomizer similar to the existing foreground placement randomizer, but placed along the XZ plane instead of the XY plane.

Reply:
👍